File: parens.js
function balanced_brackets(s) {
  var counter = 0;
  var index = 0;
  while (counter >=0 && index < s.length) {
    if (s[index] === '(') {
      counter++;
    }
    else if (s[index] === ')') {
      counter--;
    }
    index++;
  }   
  if (counter === 0) {
    console.log('Balanced');
  } else {
    console.log('Unbalanced');
  }
}

File: parens.rb
class Challenge

  def self.parens(string)
    counter = 0
    index = 0
    while counter >=0 && index < string.length
      if string[index] == '('
        counter += 1
      elsif string[index] == ')'
        counter -= 1
      end
      index += 1
    end    
    if counter == 0
      puts 'yes'
    else
      puts 'no'
    end

  end
end


Challenge.parens('()(A(X))')
Challenge.parens(')(')
Challenge.parens('(()))()')
Challenge.parens('(()()))')
